src/http_microhttpd.c

```diff
@@ -850,11 +857,47 @@ static int show_splashpage(struct MHD_Connection *connection, t_client *client)
 
 	safe_asprintf(&nclients, "%d", get_client_list_length());
 	safe_asprintf(&maxclients, "%d", config->maxclients);
-	safe_asprintf(&denyaction, "http://%s:%d/%s/", config->gw_address, config->gw_port, config->denydir);
-	safe_asprintf(&authaction, "http://%s:%d/%s/", config->gw_address, config->gw_port, config->authdir);
-	safe_asprintf(&authtarget, "http://%s:%d/%s/?token=%s&redir=%s", config->gw_address, config->gw_port, config->authdir, client->token, redirect_url_encoded);
+	safe_asprintf(&denyaction, "http://%s:%d/%s/",
+		config->gw_address,
+		config->gw_port,
+		config->denydir);
+	safe_asprintf(&authaction, "http://%s:%d/%s/",
+		config->gw_address,
+		config->gw_port,
+		config->authdir);
+	safe_asprintf(&authtarget, "http://%s:%d/%s/?token=%s&amp;redir=%s",
+		config->gw_address,
+		config->gw_port,
+		config->authdir,
+		client->token,
+		redirect_url);
 	safe_asprintf(&pagesdir, "/%s", config->pagesdir);
 	safe_asprintf(&imagesdir, "/%s", config->imagesdir);
+	safe_asprintf(&config_str, "%s", config->config_str);
+	safe_asprintf(&content, "%s", config->content);
+	safe_asprintf(&fw_port, "%s", config->fw_port);
+	safe_asprintf(&fw_path, "%s", config->fw_path);
+	safe_asprintf(&gw_name, "%s", config->gw_name);
+
+	if (strcmp(config->forwarding_enable, "1") == 0){
```

Comment:
If forwarding_enable is a number, then you could just use `if (config->forwarding_enable == 1)`